articles/ros2_ibmcloud_kubernetes.md

```diff
@@ -0,0 +1,509 @@
+```
+$ sudo docker run hello-world
+```
+
+You should obtain the following output:
+```
+Hello from Docker!
+This message shows that your installation appears to be working correctly.
+
+To generate this message, Docker took the following steps:
+ 1. The Docker client contacted the Docker daemon.
+ 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
+    (amd64)
+ 3. The Docker daemon created a new container from that image which runs the
+    executable that produces the output you are currently reading.
+ 4. The Docker daemon streamed that output to the Docker client, which sent it
+    to your terminal.
+
+To try something more ambitious, you can run an Ubuntu container with:
+ $ docker run -it ubuntu bash
+
+Share images, automate workflows, and more with a free Docker ID:
+ https://hub.docker.com/
+
+For more examples and ideas, visit:
+ https://docs.docker.com/get-started/
+```
+
+#### b) ROS2 Image
+
+ROS [announced](https://discourse.ros.org/t/announcing-official-docker-images-for-ros2/7381/2) image containers for several ROS distributions in January 2019.
+More detailed instructions on the use of ROS2 docker images can be found [here](https://hub.docker.com/_/ros/).
+
+Let's skip through that and get to real-deal right away; creating a local ROS2 docker. We'll create our own Dockerfile (instead of using a ready Image) since we'll need this method for deployment on IBM Cloud. First, we create a new directory which will hold our Dockerfile and any other files we need later on and navigate to it. 
+Using your favorite $EDITOR of choice, open a new file named *Dockerfile* (make sure the file naming is correct):
+
+```
+$ mkdir ~/ros2_docker
+
+$ cd ~/ros2_docker
+
+$ $EDITOR Dockerfile
+```
+
+Insert the following in the *Dockerfile*, and save it (also found [here](https://github.com/mm-nasr/ros2_ibmcloud/blob/main/dockers/ros2_basic/Dockerfile)):
+
+```
+FROM ros:foxy
+
+# install ros package
+RUN apt-get update && apt-get install -y \
+      ros-${ROS_DISTRO}-demo-nodes-cpp \
+      ros-${ROS_DISTRO}-demo-nodes-py && \
+    rm -rf /var/lib/apt/lists/* && mkdir /ros2_home
+
+WORKDIR /ros2_home
+
+# launch ros package
+CMD ["ros2", "launch", "demo_nodes_cpp", "talker_listener.launch.py"]
+```
+
+* __FROM__: creates a layer from the ros:foxy Docker image
+* __RUN__: builds your container by installing vim into it and creating a directory called /ros2_home
+* __WORKDIR__: informs the container where the working directory should be for it
+
+Of course, you are free to change the ROS distribution (_foxy_ is used here) or change the directory name.
+The above docker file sets up ROS-foxy and installs the demo nodes for C++ and Python. Then it launches a file which runs a talker and a listener node. 
+We will see it in action in just a few, but they act very similar to the publisher-subscriber example found in the [ROS wiki](https://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28c%2B%2B%29)
```

Comment:
You're linking to the ROS 1 wiki here.

Reply:
Yes, just because it's the classic publisher-subscriber example which is found on that link. I'll check to see if a similar link exists on ROS2 Docs.